File: publicq-main/client/src/models/ip-rate-limit-update-request.ts
/**
 * Update request model for IP rate limiting settings.
 */
export interface IpRateLimitUpdateRequest {
  
  /** Whether IP rate limiting is enabled or disabled. */
  enabled: boolean;

  /**
   * A list of IP addresses (v4 and v6) that are exempt from rate limiting.
   */
  ipWhitelist: string[];

  /**
   * Header used to determine the real client IP when behind a proxy.
   */
  realIpHeader: string;

  /**
   * API calls limit for the specified period.
   */
  limit: number;

  /**
   * Period for the rate limit in seconds.
   */
  periodInSeconds: number;
}

/**
 * Default IP rate limit update request values.
 */
export const defaultIpRateLimitUpdateRequest: IpRateLimitUpdateRequest = {
  enabled: true,
  ipWhitelist: [],
  realIpHeader: "X-Real-IP",
  limit: 100,
  periodInSeconds: 60
};

File: publicq-main/src/PublicQ.API/Models/Requests/IpRateLimitUpdateRequest.cs
namespace PublicQ.API.Models.Requests;

/// <summary>
/// Update request model for IP rate limiting settings.
/// </summary>
public class IpRateLimitUpdateRequest
{
    /// <summary>
    /// Indicates whether IP rate limiting is enabled.
    /// </summary>
    public bool Enabled { get; set; }
    
    /// <summary>
    /// A list of IP addresses (v4 and v6) that are exempt from rate limiting.
    /// </summary>
    public List<string> IpWhitelist { get; set; } = [];
    
    /// <summary>
    /// Header used to determine the real client IP when behind a proxy.
    /// </summary>
    public string RealIpHeader { get; set; } = "X-Real-IP";
    
    /// <summary>
    /// API calls limit for the specified period.
    /// </summary>
    public int Limit { get; set; }
    
    /// <summary>
    /// Period for the rate limit in seconds.
    /// </summary>
    public int PeriodInSeconds { get; set; } = default!;
}